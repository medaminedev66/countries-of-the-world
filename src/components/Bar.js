import React from 'react';
import { useHistory } from 'react-router';
import { PropTypes } from 'prop-types';

function Bar(props) {
  const history = useHistory();
  const handler = () => {
    history.push('/');
  };
  const { title } = props;
  return (
    <div className="bar">
      <button type="button" onClick={handler}>
        &lt;
      </button>
      <h2>{title}</h2>
    </div>
  );
}

Bar.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Bar;
