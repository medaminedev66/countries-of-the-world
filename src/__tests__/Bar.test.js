import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Bar from '../components/Bar';

describe('test the Quote components', () => {
  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Bar />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const { queryByTestId } = render(
      <Router>
        <Bar />
      </Router>,
    );
    expect(queryByTestId('bar')).toBeTruthy();
  });
});
