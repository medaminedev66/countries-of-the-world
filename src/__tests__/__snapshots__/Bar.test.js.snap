// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the Quote components matches the snapshot 1`] = `
<div
  className="bar"
  data-testid="bar"
>
  <button
    onClick={[Function]}
    type="button"
  >
    &lt;
  </button>
  <h2 />
</div>
`;
